### PR TITLE
Contributing files - spanish translation

### PR DESCRIPTION
**Pull request by jmanubarreiro**
Spanish translation of:

- backwards-compatibility.rst
- cakephp-coding-conventions.rst
- code.rst
- documentation.rst
- tickets.rst

### CODE REVIEW

File: es/contributing/code.rst [hunk 1]
```diff
@@ -1,15 +1,154 @@
...
+    git checkout -b ticket-1234 upstream/master
+
+    # arreglando un error en 2.x
+    git fetch upstream
+    git checkout -b ticket-1234 upstream/2.x
+
+.. tip::
+
+    Usa un nombre descriptivo para tu rama, referenciar el ticket o nombre de la
+	característica es una buena convención. P.ej. ticket-1234, neuva-funcionalidad
```
Comment (markstory) [inline es/contributing/code.rst]: This line has extra indentation that may not render well.

Comment (jmanubarreiro) [inline es/contributing/code.rst]: indentation fixed

**Comment by jmanubarreiro** [comment]
@markstory ,  thanks for the review.

I have a doubt with the last ci check failed...

For what i see, the only error that there seems to be is the reference to the label 'running-tests'  of the section with the same name in testing.rst.

The spanish translation of this page doesn`t exist yet, so the tag neither.

Which is the best way to proceed?

This pull request can be merged with this error?
testing.rst has to be translated mandatorily before?

File: es/contributing/cakephp-coding-conventions.rst [hunk 2]
```diff
@@ -1,15 +1,618 @@
...
+Si el nombre de una constante se compone de varias palabras deberán ir separadas
+por guiones bajos, por ejemplo::
+
+    define('NOMBRE_DE_CONSTANTE_LARGO', 2);
+
+Cuidado al usar empty()/isset()
+===================================
+
+Aunque ``empty()`` es una función sencilla de utilizar, puede enmascarar errores
+y causar efectos inintencionados cuando se presentan ``'0'`` y ``0``. Cuando
```
Comment (albertcansado) [inline es/contributing/cakephp-coding-conventions.rst]: `inintencionados cuando se presentan` -> efectos accidentales cuando se usa con ...

File: es/contributing/cakephp-coding-conventions.rst [hunk 3]
```diff
@@ -1,15 +1,618 @@
...
+    }
+
+Cuando se trabaja con arrays, es mejor hacer ``merge`` de valores por defecto
+en vez de hacer comprobaciones con ``empty()``. Haciendo ``merge`` de valores
+por defecto puedes asegurarte de que las claves necesarias están definidas::
+
+    function doWork(array $array)
+    {
+        // Hacer merge de valor por defecto para eliminar la necesidad
+		// de comprobaciones empty
```
Comment (albertcansado) [inline es/contributing/cakephp-coding-conventions.rst]: indent :)

File: es/contributing/code.rst [hunk 4]
```diff
@@ -1,15 +1,154 @@
-Code
-####
+Código
+######
+
+Parches y *pull requests* son una manera genial de contribuir con código a CakePHP. 
+Los *Pull requests* pueden ser creados en Github, preferiblemente a los archivos de
+parches en comentarios de tickets.
```
Comment (albertcansado) [inline es/contributing/code.rst]: en `los` comentarios

File: es/contributing/code.rst [hunk 5]
```diff
@@ -1,15 +1,154 @@
...
+    git checkout -b ticket-1234 upstream/master
+
+    # arreglando un error en 2.x
+    git fetch upstream
+    git checkout -b ticket-1234 upstream/2.x
+
+.. tip::
+
+    Usa un nombre descriptivo para tu rama, referenciar el ticket o nombre de la
+    característica es una buena convención. P.ej. ticket-1234, neuva-funcionalidad
```
Comment (albertcansado) [inline es/contributing/code.rst]: `neuva-` -> `nueva`

File: es/contributing/code.rst [hunk 6]
```diff
@@ -1,15 +1,154 @@
...
+    git fetch upstream
+    git merge upstream/master
+    git checkout <nombre_rama>
+    git rebase master
+
+Esto buscará y hará *merge* de cualquier cambio que haya sucedido en CakePHP desde que
+empezaste. Entonces ejecutará *rebase* o replicará tus cambios en el *top* del
+actual código. 
+
+Puede que encuentres algún conflicto durante el *rebase*. Si éste finaliza
```
Comment (albertcansado) [inline es/contributing/code.rst]: `este`

File: es/contributing/code.rst [hunk 7]
```diff
@@ -1,15 +1,154 @@
...
+
+    git push --force origin <nombre-rama>
+
+Una vez tu rama esté en GitHub puedes enviar un *pull request* en GitHub.
+
+Seleccionar donde harán el *merge* tus cambios
+----------------------------------------------
+
+Cuando hagas *pull requests* deberás asegurarte de seleccionar la rama correcta
+como base ya que no podrás editarla una vez sea creada.
```
Comment (albertcansado) [inline es/contributing/code.rst]: ~~sea~~ creada

File: es/contributing/documentation.rst [hunk 8]
```diff
@@ -1,29 +1,493 @@
...
-comiences a colaborar.
-
-La documentación de CakePHP cuenta con
-`integración continua <https://es.wikipedia.org/wiki/Integraci%C3%B3n_continua>`_,
-así que puedes er el status de `varias ejecuciones <https://ci.cakephp.org>`_
-en el servidor de Jenkins cuando quieras.
+Contribuir con la documentación es fácil. Los archivos están hospedados
+en https://github.com/cakephp/docs. Siéntete libre de hacer un *fork* del
+repositorio, añadir tus cambios, mejoras, traducciones y comenzar a ayudar
+a través de un nuevo *pull request*. Puedes también editar los archivos de manera
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: `También puedes`

File: es/contributing/documentation.rst [hunk 9]
```diff
@@ -1,29 +1,493 @@
...
+---------------
+
+Las líneas de texto deberían medir como máximo 40 caracteres. Las únicas
+excepciones son URLs largas y fragmentos de código.
+
+Cabeceras y secciones
+---------------------
+
+Las cabeceras de las secciones se crean subrayando el título con caracteres de 
+puntuación. El subrayado deberá de ser por lo menos tan largo como el texto.
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: deberá ~~de~~ ser

File: es/contributing/documentation.rst [hunk 10]
```diff
@@ -1,29 +1,493 @@
...
+- ``^`` Se utiliza para los títulos de sub-sub-subsecciones.
+
+Los encabezados no deben anidarse con más de 5 niveles de profundidad y deben
+estar precedidos y seguidos por una línea en blanco.
+
+Párrafos
+--------
+
+Párrafos son simplemente bloques de texto con todas las líneas al mismo nivel de
+indexación. Los párrafos deben de separarse por al menos una línea vacía.
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: deben ~~de~~ separarse

File: es/contributing/documentation.rst [hunk 11]
```diff
@@ -1,29 +1,493 @@
...
+  * ````cascadeCallbacks````, ````true````, ````id````,
+    ````PagesController````, ````config()````, etc.
+
+Si aparecen asteriscos o acentos graves en el texto y pueden ser confundidos con
+los delimitadores de marcado habrá que escaparlos con *\\*.
+
+Los marcadores en línea tienen algunas restricciones:
+
+* **No pueden** estar anidados.
+* El contenido no puede empezar o a cabar con espacios en blanco: ``* texto*``
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: `a cabar` -> `acabar`

File: es/contributing/documentation.rst [hunk 12]
```diff
@@ -1,29 +1,493 @@
...
+* El contenido debe separarse del resto del texto por caracteres que no sean
+  palabras. Utiliza *\\* para escapar un espacio y solucionarlo: ``onelong\ *bolded*\ word``.
+
+Listas
+------
+
+El etiquetado de listas es muy parecido a *Markdown*. Las listas no ordenadas se
+indican empezando una línea con un asterisco y un espacio.
+
+Las listas enumeradas pueden crearse con enumaraciones o ``#`` para auto enumaración::
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: ``enumaranciones`` -> ``enumeraciones`` && enum**e**ración

File: es/contributing/documentation.rst [hunk 13]
```diff
@@ -1,29 +1,493 @@
...
+    referencia a ``core-helpers/form``. Si la referencia fuese ``:doc:`/core-helpers```
+    el enlace sería siempre a ``/core-helpers`` sin importar donde se utilice.
+
+Enlaces a referencias cruzadas
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. rst:role:: ref
+
+    Puedes hacer referncia cruzada a cualquier título de cualquier documento 
+    usando la función ``:ref:``. Los enlaces a etiquetas de destino deben de ser
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: deben ~~de~~ ser

File: es/contributing/documentation.rst [hunk 14]
```diff
@@ -1,29 +1,493 @@
...
+    El uso más habitual de etiquetas es encima de un título. Ejemplo::
+    
+        .. _nombre-etiqueta:
+	
+	Título sección
+	--------------
+	
+	Resto del contenido.
+
+    En otro sitió podrías enlazar a la sección de arriba usando ``:ref:`nombre-etiqueta```.
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: ``sitió`` -> ``sitio``

File: es/contributing/documentation.rst [hunk 15]
```diff
@@ -1,29 +1,493 @@
...
+herramientas de referenciación cruzada.
+
+Describir clases y constructores
+--------------------------------
+
+Cada directiva introduce el contenido del índice y/o índice del *namespace*.
+
+.. rst:directive:: .. php:global:: nombre
+
+   Esta directiva declara una nueva variable PHP glopabl.
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: ``glopabl`` -> ``global``

File: es/contributing/documentation.rst [hunk 16]
```diff
@@ -1,29 +1,493 @@
...
+
+.. rst:directive:: .. php:function:: nombre(firma)
+
+   Define una nueva función global fuera de una clase.
+
+.. rst:directive:: .. php:const:: nombre
+
+   Esta directiva declara una nueva constante PHP, puedes usarla también anidada
+   dentro de una directiva de clas para crear constantes de clase.
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: ``clas`` -> ``clase``

File: es/contributing/documentation.rst [hunk 17]
```diff
@@ -1,29 +1,493 @@
...
+        .. php:class:: MyClass
+
+            Descripción de la clase
+
+           .. php:method:: method($argument)
+
+           Descripción del método
+
+
+   Atributos, métodso y constantes no necesitan estar anidados, pueden seguir
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: ``métodso`` -> ``métodos``

File: es/contributing/documentation.rst [hunk 18]
```diff
@@ -1,29 +1,493 @@
...
+           Esto es una instancia de método.
+
+.. rst:directive:: .. php:staticmethod:: ClassName::nombreMetodo(firma)
+
+    Describe un método estático, sus argumentos, salida y excepciones,
+    ver :rst:dir:`php:method` para opciones.
+
+.. rst:directive:: .. php:attr:: nombre
+
+   Describe na propiedad/atributo en una clase.
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: ``na`` -> ``una``

File: es/contributing/documentation.rst [hunk 19]
```diff
@@ -1,29 +1,493 @@
...
+
+Referencias cruzadas
+~~~~~~~~~~~~~~~~~~~~
+
+Los siguientes *roles* hacen referencia a objetos PHP y los enlaces son generados
+si se encuentra una directiva que coincida:
+
+.. rst:role:: php:func
+
+   Referencia a función PHP.
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: a ``una`` función

File: es/contributing/documentation.rst [hunk 20]
```diff
@@ -1,29 +1,493 @@
...
+
+.. rst:role:: php:global
+
+   Referencia a varaible global cuyo nombre tiene prefijo ``$``.
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: a ``una`` ``variable``

File: es/contributing/documentation.rst [hunk 21]
```diff
@@ -1,29 +1,493 @@
...
+
+.. rst:role:: php:const
+
+   Referencia tanto a una constante global como a una de clase. Las constantes
+   de clase deberán ir precedidas por la clase que las contengan::
```
Comment (albertcansado) [inline es/contributing/documentation.rst]: ``contengan`` -> ``contenga``

File: es/contributing/tickets.rst [hunk 22]
```diff
@@ -1,15 +1,51 @@
...
+Aportar *feedback* y ayudar a la comunidad en la forma de tickets es una parte
+extremadamente importante en el proceso de desarrollo de CakePHP. Todos los 
+tickets de CakePHP están alojados en `GitHub <https://github.com/cakephp/cakephp/issues>`_.
 
-    Por favor, siéntase libre de enviarnos un pull request en
-    `Github <https://github.com/cakephp/docs>`_ o utilizar el botón **Improve this Doc** para proponer directamente los cambios.
+Reportar errores
+================
+
+Reportes de errores bien escritos son de mucha ayuda. Para ello hay una serie de
```
Comment (albertcansado) [inline es/contributing/tickets.rst]: ``Los`` reportes

File: es/contributing/tickets.rst [hunk 23]
```diff
@@ -1,15 +1,51 @@
...
-    `Github <https://github.com/cakephp/docs>`_ o utilizar el botón **Improve this Doc** para proponer directamente los cambios.
+Reportar errores
+================
+
+Reportes de errores bien escritos son de mucha ayuda. Para ello hay una serie de 
+pasos que ayudan a crear el mejor reporte de error posible:
+
+* **Correcto**: Por favor, `busca tickets <https://github.com/cakephp/cakephp/search?q=it+is+broken&ref=cmdform&type=Issues>`_
+  similares que ya existan y asegúrate de que nadie haya reportado ya tu problema 
+  o de que no haya sido arreglado en el repositorio.
```
Comment (albertcansado) [inline es/contributing/tickets.rst]: o ~~de~~ que no haya

File: es/contributing/tickets.rst [hunk 24]
```diff
@@ -1,15 +1,51 @@
...
+* **Correcto**: Por favor, incluye instrucciones detalladas de **cómo reproducir el error**. 
+  Ésto podría estar escrito en el formato de caso de prueba o con un
```
Comment (albertcansado) [inline es/contributing/tickets.rst]: ``Èsto`` -> ``Esto``

File: es/contributing/backwards-compatibility.rst [hunk 25]
```diff
@@ -1,11 +1,177 @@
...
+
+Para cada liberación ``major`` y ``minor`` el equipo de CakePHP facilitará una
+guía de migración. Estas guías explican las nuevas funcionaliades y cualquier
+ruptura de compatibilidad que haya en cada lanzamiento. Pueden encontrarse en
+la sección :doc:`/appendices` del ``cookbook``.
+
+Usar CakePHP
+============
+
+Si estás desarrollando tu aplicación con CakePHP las siguientes líneas guía
```
Comment (jmanubarreiro) [inline es/contributing/backwards-compatibility.rst]: sip, mejor pautas

File: es/contributing/cakephp-coding-conventions.rst [hunk 26]
```diff
@@ -1,15 +1,618 @@
...
+
+Para comprobar si una variables es ``null`` se recomienda utilizar comprobación
+estricta::
+
+    if ($value === null) {
+        // ...
+    }
+
+El valor contra el que se va a realizar la comparación deberá ir en el lado derecho
+de ésta::
```
Comment (jmanubarreiro) [inline es/contributing/cakephp-coding-conventions.rst]: aún no me acostumbro a este cambio de la norma...

**Comment by jmanubarreiro** [comment]
Seeing the number of errors with this pull i'll close it and make others with a file by request

**Pull request closed without merging**